apis/python/src/tiledbsoma/types.py

```diff
@@ -30,11 +30,22 @@
 DenseCoordinates = Union[int, slice, pa.Array]
 DenseNdCoordinates = Tuple[DenseCoordinates, ...]
 
+# TODO: add support for non-ints once the libtiledbsoma SOMAReader class has supports
+# for non-ints. See also:
+# https://github.com/single-cell-data/TileDB-SOMA/issues/418
+# https://github.com/single-cell-data/TileDB-SOMA/issues/419
+#
 # Note: we intentionally use `Union[None, ...]` in place of `Optional[...]` since
 # we choose to emphasize that the argument-slots this is used in are not "optional"
 # arguments -- they're required argments, which can take the `None` value.
+SparseIndexedDataFrameCoordinate = Union[
```

Comment:
This may be a naive question (but here goes :-):  how is the Union of a Union different than flattening it into a single Union?

I.e., why not remove one level, as in:
```
SparseIndexedDataFrameCoordinate = Union[
    None, int, slice, Sequence[int], pa.Array, pa.ChunkedArray, np.ndarray
]
```

Reply:
Also, NumPy typing should use their explicit typing support IMHO.   See the [docs](https://numpy.org/doc/stable/reference/typing.html?highlight=data%20type%20api)

The difference here is that you would first 
> import numpy.typing as npt

And then declare this as:
```
SparseIndexedDataFrameCoordinate = Union[
    None, int, slice, Sequence[int], pa.Array, pa.ChunkedArray, npt.NDArray[np.int_]
]
```

If you don't want to bite this off in this PR, just capture as an open issue. I would understand!

Reply:
#517 -- thank you!